packages/data-broker/src/services/dhis/DhisInputSchemeResolvingApiProxy.js

```diff
@@ -192,6 +185,63 @@ export class DhisInputSchemeResolvingApiProxy {
     return modifiedQuery;
   };
 
+  /**
+   * @param response {*}
+   * @param dataElementCodes {*}
+   * @returns {*}
+   * @private
+   */
+  translateDataElementIdsToCodesInResponse = async (response, dataElementCodes) => {
+    const dataElementIdToCode = {};
+
+    const dataElements = await this.models.dataSource.find({
+      code: dataElementCodes,
+      type: 'dataElement',
+    });
+
+    for (const dataElement of dataElements) {
+      if (dataElement.config.dhisId) {
+        dataElementIdToCode[dataElement.config.dhisId] = dataElement.code;
+      }
+    }
+
+    return translateElementKeysInEventAnalytics(response, dataElementIdToCode);
+  };
+
+  /**
+   * @param response {*}
+   * @returns {*}
+   * @private
+   */
+  translateOrgUnitIdsToCodesInResponse = async response => {
+    if (!response.rows.length) return response;
+    const newRows = [];
+
+    const orgUnitIdIndex = response.headers.findIndex(({ name }) => name === 'ou');
+    const orgUnitCodeIndex = response.headers.findIndex(({ name }) => name === 'oucode');
+    if (!orgUnitIdIndex || !orgUnitCodeIndex)
```

Comment:
```suggestion
    if (orgUnitIdIndex === -1 || orgUnitCodeIndex === -1)
```

Reply:
Eek! I should have written a test case for it!